packages/build/src/install/missing.js

```diff
@@ -80,24 +81,30 @@ const AUTO_PLUGINS_PACKAGE_JSON = {
 }
 
 // External plugins must be installed either in the UI or in `package.json`.
-// A third way is deprecated: adding it to `netlify.toml` but not in
-// `package.json`. In that case, the plugin will be automatically installed
-// like UI plugins.
-// We still support this for backward compatibility but print a warning on each
-// build (even if the plugin was installed in a previous build).
-// We deprecate this third way because:
-//  - having fewer ways of installing plugins is simpler
-//  - using `package.json` is faster and more reliable
-const warnOnConfigOnlyPlugins = function ({ pluginsOptions, logs }) {
-  const packages = pluginsOptions.filter(isConfigOnlyPlugin).map(getPackageName)
+// A third way is supported but undocumented: adding it to `netlify.toml` but
+// not in `package.json`. In that case, the plugin will be automatically
+// installed.
+// This is only supported for plugins listed in `plugins.json`. Otherwise,
+// we should fail the build. At the moment, we only print a warning for
+// backward compatibility.
+const warnOnConfigOnlyPlugins = function ({ pluginsOptions, featureFlags, logs }) {
+  const packages = pluginsOptions
+    .filter(({ loadedFrom, origin, expectedVersion }) =>
+      isConfigOnlyPlugin({ loadedFrom, origin, expectedVersion, featureFlags }),
+    )
+    .map(getPackageName)
   if (packages.length === 0) {
     return
   }
 
   logConfigOnlyPlugins(logs, packages)
 }
 
-const isConfigOnlyPlugin = function ({ loadedFrom, origin }) {
+const isConfigOnlyPlugin = function ({ loadedFrom, origin, expectedVersion, featureFlags }) {
+  if (featureFlags.new_plugins_install) {
+    return expectedVersion === 'latest'
```

Comment:
This means the plugin is not in our `plugins.json` and was not in the site's `package.json`. We print a warning then.

Reply:
Linking relevant code:
https://github.com/netlify/build/blob/dae1167d350f23da54b729e2dc11db9b908b28e5/packages/build/src/plugins/expected_version.js#L36